src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,67 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int INDEX_ARRAY_FROM_THE_NUMBER = 1;
+    private static final String SUPPLY_NAME = "supply";
+    private static final String BUY_NAME = "buy";
+    private static final String SEPARATOR = ",";
+    private static final String RESULT_NAME = "result";
 
     public void getStatistic(String fromFileName, String toFileName) {
+        String inputCsv = fileReader(fromFileName);
+        fileWriter(findSupplyAndBuy(inputCsv),toFileName);
+    }
+
+    private String findSupplyAndBuy(String text) {
+        String result = "";
+        int sumSupply = 0;
+        int sumBuy = 0;
+        String[] arrayFullText = text.split(System.lineSeparator());
+        for (String lineArray : arrayFullText) {
+            String[] line = lineArray.split(SEPARATOR);
+            if (lineArray.contains(SUPPLY_NAME)) {
+                sumSupply += Integer.parseInt(line[INDEX_ARRAY_FROM_THE_NUMBER]);
+            } else {
+                sumBuy += Integer.parseInt(line[INDEX_ARRAY_FROM_THE_NUMBER]);
+            }
+        }
+        result += SUPPLY_NAME + SEPARATOR + sumSupply + System.lineSeparator();
+        result += BUY_NAME + SEPARATOR + sumBuy + System.lineSeparator();
+        result += RESULT_NAME + SEPARATOR + (sumSupply - sumBuy);
+        return result;
+    }
+
+    private String fileReader(String fromFileName) {
+        File file = new File(fromFileName);
+        StringBuilder stringBuilder = new StringBuilder();
+        try {
+            BufferedReader bufferedReader = new BufferedReader(new FileReader(file));
+            String value = bufferedReader.readLine();
```

Comment:
<img width="953" alt="image" src="https://user-images.githubusercontent.com/99619831/182333281-9fff0a8a-d6e4-4a0d-a7ff-6ec3fe92312d.png">

please go through checklist points and refactor your code